syntax = "proto3";
package proto;
import "core.proto";

option go_package = "github.com/anyuan-chen/record/proto/pkg/image_processing_pb";

message Images {
    repeated ImageURL images = 1;
    int64 rowCount = 2;
    int64 colCount = 3;
    int64 targetSize = 4;
}
message ImageURL {
    string imageURL = 1;
}

service ImageProcessor {
    rpc GetCollage (Images) returns (Image);
}

// python3 -m grpc_tools.protoc -I=./ --python_out=./pkg/image_processing_pb \
// --grpc_python_out=./pkg/image_processing_pb image_processing.proto

// protoc -I=./ --go_out=./pkg/image_processing_pb --go_opt=paths=source_relative \
// --go-grpc_out=./pkg/image_processing_pb --go-grpc_opt=paths=source_relative \
// image_processing.proto 